sstables_loader.cc

```diff
@@ -188,9 +193,26 @@ class tablet_sstable_streamer : public sstable_streamer {
         // FIXME: fully contained sstables can be optimized.
         return stream_sstables(pr, std::move(sstables), std::move(on_streamed));
     }
+
+    bool tablet_in_scope(locator::tablet_id) const;
 };
 
 inet_address_vector_replica_set sstable_streamer::get_endpoints(const dht::token& token) const {
+    return get_all_endpoints(token) | std::views::filter([&topo = _erm->get_topology(), scope = _stream_scope] (const auto& ep) {
+        switch (scope) {
+        case stream_scope::all:
+            return true;
+        case stream_scope::dc:
+            return topo.get_datacenter(ep) == topo.get_datacenter();
+        case stream_scope::rack:
+            return topo.get_rack(ep) == topo.get_rack();
+        case stream_scope::node:
+            return topo.is_me(ep);
+        }
+    }) | std::ranges::to<inet_address_vector_replica_set>();
+}
+
+inet_address_vector_replica_set sstable_streamer::get_all_endpoints(const dht::token& token) const {
     if (_primary_replica_only) {
         return get_primary_endpoints(token);
```

Comment:
1. `sstable_streamer::get_primary_endpoints` is out of sync with `tablet_map::get_primary_replica` (preexisting issue, but we should fix it)
2. `_primary_replica_only` should interact with scope such that it will return the primary replica within that scope.  We can use `tablet_map::get_primary_replica` for scope `all`, and we already have `tablet_map::get_primary_replica_within_dc` for the `dc` scope.  We can have a similar entry point for `tablet_map::get_primary_replica_within_rack`.

As for the `node` scope, I suggest we return an error if both `--primary-replica-only` and `--scope=node` are used, to keep things simple.

Reply:
1. Is there an issue for that?
2. I think I better reject primary-replica-only=true with any scope but all, let's clarify it as a followup, it's been causing questions since the scope was introduced at all,

Reply:
> 1. Is there an issue for that?

Not that I know of.  Let's open one.

> 2. I think I better reject primary-replica-only=true with any scope but all, let's clarify it as a followup, it's been causing questions since the scope was introduced at all,

okay for now

Reply:
#21883